demo/xcb_cairo/nuklear_xcb.h

```diff
@@ -0,0 +1,852 @@
+/*****************************************************************************
+ *
+ * Nuklear XCB/Cairo Render Backend - v0.0.2
+ * Copyright 2021 Richard Gill
+ * 
+ * Grabbed and adapted from https://github.com/griebd/nuklear_xcb
+ * Copyright 2017 Adriano Grieb
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ ****************************************************************************/
+
+/*****************************************************************************
+ *
+ *                                API
+ *
+ ****************************************************************************/
+
+#ifndef NK_XCB_CAIRO_H
+#define NK_XCB_CAIRO_H
+
+/* just to make my IDE happy */
+#ifndef NK_NUKLEAR_H_
+#include "../../nuklear.h"
+#endif
```

Comment:
Hmm, I would pref we not have IDE specific "hacks", but I'm note quite sure what the "ideomatic" way of depending on the nuklear header here is. Maybe:
```c
#undef NK_IMPLEMENTATION
#include "../../nuklear.h"
```

Still feels a little gross